lib/services/api/api_service.ts

```diff
@@ -128,12 +129,12 @@ export class SNApiService extends PureService {
           messages.API_MESSAGE_GENERIC_INVALID_LOGIN
         );
       });
-    return response;
+    return response as KeyParamsResponse;
   }
 
   async register(email: string, serverPassword: string, keyParams: SNRootKeyParams) {
     if (this.registering) {
-      return this.createErrorResponse(messages.API_MESSAGE_REGISTRATION_IN_PROGRESS);
+      return this.createErrorResponse(messages.API_MESSAGE_REGISTRATION_IN_PROGRESS) as RegistrationResponse;
```

Comment:
This works because `RegistrationResponse`'s additional properties are all optional, however the day we stray from that unwritten rule we'll crash clients by feeding them false information. Perhaps a comment on `RegistrationResponse` could be enough but there should be something preventing `TypeError`s here.

Reply:
For now it would seem that all response values will be optional if the type is &'ed with HttpResponse. This is because HttpResponse could always have an error, in which case all values would be undefined. Perhaps if we called it `RegistrationSuccessResponse`, we would then have to start thinking about non-optionality.

Reply:
That's fine but we should document it in the type declaration so we don't forget and add a non-optional property in the future.